account_comment_template/tests/test_account_move_report.py

```diff
@@ -1,44 +1,64 @@
 # Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo.tests.common import TransactionCase
+from odoo.tests.common import Form, TransactionCase
 
 
 class TestAccountInvoiceReport(TransactionCase):
-    at_install = False
-    post_install = True
-
     def setUp(self):
-        super(TestAccountInvoiceReport, self).setUp()
+        super().setUp()
+        self.company = self.env.ref("base.main_company")
         self.base_comment_model = self.env["base.comment.template"]
+        self.move_obj = self.env.ref("account.model_account_move")
         self.before_comment = self._create_comment("before_lines")
         self.after_comment = self._create_comment("after_lines")
         self.partner = self.env["res.partner"].create({"name": "Partner Test"})
-        self.invoice_model = self.env["account.move"]
-        self.invoice = self.invoice_model.create(
+        self.partner.base_comment_template_ids = [
+            (4, self.before_comment.id),
+            (4, self.after_comment.id),
+        ]
+        self.income_account = self.env["account.account"].search(
+            [("user_type_id.name", "=", "Income")], limit=1
+        )
+        self.journal_sale = self.env["account.journal"].create(
             {
-                # "move_type": "out_invoice",
-                # "journal_id":
-                "partner_id": self.partner.id,
-                "comment_template1_id": self.before_comment.id,
-                "comment_template2_id": self.after_comment.id,
+                "name": "Test journal sale",
+                "code": "TST-JRNL-S",
+                "type": "sale",
+                "company_id": self.company.id,
             }
         )
+        move_form = self._create_invoice()
+        self.invoice = move_form.save()
 
-        self.invoice._set_note1()
-        self.invoice._set_note2()
+    def _create_invoice(self):
+        move_form = Form(
+            self.env["account.move"].with_context(default_type="out_invoice")
+        )
+        move_form.partner_id = self.partner
+        move_form.journal_id = self.journal_sale
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.name = "test"
+            line_form.quantity = 1.0
+            line_form.price_unit = 100
+            line_form.account_id = self.income_account
+        return move_form
 
     def _create_comment(self, position):
         return self.base_comment_model.create(
             {
                 "name": "Comment " + position,
+                "company_id": self.company.id,
                 "position": position,
                 "text": "Text " + position,
+                "model_ids": [(6, 0, self.move_obj.ids)],
             }
         )
 
-    def test_comments_in_invoice(self):
+    def test_comments_in_invoice_report(self):
+        self.invoice._compute_comment_template_ids()
```

Comment:
This would not be necessary after the fix from https://github.com/OCA/reporting-engine/pull/509

It actually tricks the test and is missing a case that happens in the UI